src/core/player.js

```diff
@@ -1,71 +1,126 @@
-import { Assets, Sprite, Application } from "pixi.js";
+import { Assets, Sprite } from "pixi.js";
+import { bulletProjectile } from "./bullet";
 
 export class tankPlayer {
-    constructor(playerX, playerY){
-        // setup the x, y co-ordinates
-        this.playerX = playerX;
-        this.playerY = playerY;
-        this.playerSpeed = 5;
-        this.keys = {}; // dict setup
-    }
+  constructor(playerX, playerY, app) {
+    // setup the x, y co-ordinates
+    this.app = app;
+    this.playerX = playerX;
+    this.playerY = playerY;
+    this.playerSpeed = 5;
+    this.keys = {}; // dict setup
+    this.bullets = [];
+  }
 
-    async initialiseSprite(){
-        // load texture of player, and convert into sprite.
-        const texture = await Assets.load('assets/images/tank.png'); // 'await' keyword used for asynchronous texture loading
-        const sprite = Sprite.from(texture);
-        sprite.anchor.set(0.5, 0.5); // set anchor point to centre of the sprite
-
-        // resizing the texture
-        const desiredWidth = 150;
-        const desiredHeight = 105;
-        sprite.scale.set(desiredWidth / texture.width, desiredHeight / texture.height); // set the scale so there is no sprite distortion
-
-        // initialise x, y to arguements passed through via constructor
-        sprite.x = this.playerX;
-        sprite.y = this.playerY;
-        this.sprite = sprite;
-    }
+  getX() {
+    return this.playerX;
+  }
+
+  getY() {
+    return this.playerY;
+  }
+
+  async createBullet() {
+    const bullet = new bulletProjectile(this.playerX, this.playerY, this.app);
+    await bullet.initialiseSprite();
+    this.app.stage.addChild(bullet.getSprite());
+    this.addBulletToBullets(bullet);
+  }
 
-    addToStage(app) {
-        app.stage.addChild(this.sprite);
+  updateBullets() {
+    for (let i = 0; i < this.getBulletsList().length; i++) {
+      const projectile = this.getBulletsList()[i];
+      projectile.applyGravityToVerticalMotion();
+      projectile.updateBullet();
+
+      // check if bullet has gone off the screen, if it has, then it will be deleted. 
+      if (projectile.getX() > (this.app.canvas.width + 20) || projectile.getX() < 0) {
+        this.app.stage.removeChild(projectile);
+        this.getBulletsList().splice(i, 1);
+      }
     }
+  }
+
+  getBulletsList() {
```

Comment:
wtf we need this for?

Reply:
think i just used this for testing low key, i'll get rid of it